src/orchestrator-process/src/lib.rs

```diff
@@ -350,6 +352,15 @@ impl NamespacedOrchestrator for NamespacedProcessOrchestrator {
         fs::create_dir_all(&run_dir)
             .await
             .context("creating run directory")?;
+        let scratch_dir = if disk {
+            let scratch_dir = self.scratch_directory.join(&full_id);
+            fs::create_dir_all(&scratch_dir)
```

Comment:
this changes the flow so that the process orchestrator creates the scratch dir, rather than individual clusterds

Reply:
Much nicer, thanks!